shuffle/cli/src/transactions.rs

```diff
@@ -0,0 +1,290 @@
+use url::Url;
+
+// Will list the last 10 transactions and has the ability to block and stream future transactions.
+pub async fn handle(network: Url, tailing: bool) -> Result<()> {
+    let home = Home::new(get_home_path().as_path())?;
+    let address_str = fs::read_to_string(home.get_latest_address_path())?;
+    let address = AccountAddress::from_str(address_str.as_str())?;
+    let client = reqwest::Client::new();
+
+    let path = network.join(format!("accounts/{}/transactions", address).as_str())?;
+    let account_seq_num = get_account_sequence_number(&client, &network, address).await?;
+    let start = max(account_seq_num as i64 - 10, 0);
+
+    let resp = client
+        .get(path.as_str())
+        .query(&[("start", start.to_string().as_str())])
+        .query(&[("limit", "10")])
+        .send()
+        .await?;
+
+    let json_with_txns: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
+
+    let all_transactions = json_with_txns
+        .as_array()
+        .ok_or_else(|| anyhow::anyhow!("Failed to get transactions"))?;
+
+    for txn in all_transactions.iter() {
+        write_into(&mut io::stdout(), txn)?;
+        println!();
+        println!();
+    }
+
+    let mut old_seq_num = 0;
+    if !all_transactions.is_empty() {
+        old_seq_num = parse_txn_for_seq_num(
+            all_transactions
+                .last()
+                .ok_or_else(|| anyhow!("Couldn't get last transaction"))?,
+        )?
+    }
+    if tailing {
+        // listening for incoming transactions
+        loop {
+            thread::sleep(time::Duration::from_millis(1000));
+            let resp = get_account_transactions_response(&client, &path, old_seq_num, 100).await?;
+            let json_with_txns: serde_json::Value =
+                serde_json::from_str(resp.text().await?.as_str())?;
+            let mut txn_array = json_with_txns
+                .as_array()
+                .ok_or_else(|| anyhow!("Couldn't convert to array"))?
+                .to_vec();
+
+            // checking if there are transactions
+            if !txn_array.is_empty() {
+                let last_txn_seq_num = parse_txn_for_seq_num(
+                    txn_array
+                        .last()
+                        .ok_or_else(|| anyhow!("Couldn't get last transaction"))?,
+                )?;
+
+                if old_seq_num != 0_u64 {
+                    txn_array = normalize_txn_array(txn_array.to_vec());
+                }
+
+                old_seq_num = write_out_txns(
+                    txn_array.to_vec(),
+                    last_txn_seq_num,
+                    old_seq_num,
+                    &mut io::stdout(),
+                )?;
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn normalize_txn_array(mut arr: Vec<Value>) -> Vec<Value> {
+    // Excluding the index of the txn we've already printed out
+    arr.remove(0);
+    arr
+}
+
+pub fn write_out_txns<W: Write>(
```

Comment:
Why is this public?

Reply:
Removing pub to keep method within file.